include/mfmg/dealii/belos_traits.hpp

```diff
@@ -0,0 +1,288 @@
+/*************************************************************************
+ * Copyright (c) 2017-2019 by the mfmg authors                           *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * This file is part of the mfmg libary. mfmg is distributed under a BSD *
+ * 3-clause license. For the licensing terms see the LICENSE file in the *
+ * top-level directory                                                   *
+ *                                                                       *
+ * SPDX-License-Identifier: BSD-3-Clause                                 *
+ *************************************************************************/
+
+#ifndef MFMG_BELOS_TRAITS_HPP
+#define MFMG_BELOS_TRAITS_HPP
+
+#include <mfmg/common/exceptions.hpp>
+
+#include "multivector.hpp"
+#include <BelosMultiVecTraits.hpp>
+
+// FIXME: Belos traits are only used for compilation
+// In the runtime, Belos is only used in TraceMin and KrylovSchur algorithms,
+// and TsqrOrthoManager
+
+namespace Belos
+{
+using mfmg::ASSERT;
+
+template <typename VectorType>
+class MultiVecTraits<double, mfmg::MultiVector<VectorType>>
```

Comment:
Add a return statement to all the static member functions that need one.

Reply:
Not sure what you mean here. All static functions have `throw` in them, and do not produce warnings. Any function that has both throw and return will produce a warning.

Reply:
> Any function that has both throw and return will produce a warning.

My bad

Reply:
but we can use `[[noreturn]]`